scripts/test/Muon/phase_table_widget/phase_table_presenter_test.py

```diff
@@ -257,6 +200,38 @@ def test_that_enable_observer_calls_on_view_when_triggered(self):
                 continue
             self.assertTrue(widget.isEnabled())
 
+    def test_handle_add_phasequad_button_no_table(self):
+        self.presenter.handle_add_phasequad_button_clicked()
+        self.view.warning_popup.assert_called_once_with('Please generate a phase table first.')
+
+    def test_handle_add_phasequad_button_no_name(self):
+        self.view.set_phase_table_combo_box(["Table"])
+        self.view.enter_pair_name = mock.Mock(return_value = None)
+        self.presenter.create_phase_quad_calculation_thread = mock.MagicMock()
+
+        self.presenter.handle_add_phasequad_button_clicked()
+        self.assertEqual(self.presenter.create_phase_quad_calculation_thread.call_count, 0)
+
+    def test_handle_add_phasequad_button(self):
+        self.view.set_phase_table_combo_box(["Table"])
+        self.presenter.validate_pair_name = mock.Mock(return_value=True)
+        self.view.enter_pair_name = mock.Mock(return_value = "test")
+        self.presenter.create_phase_quad_calculation_thread = mock.MagicMock()
+
+        self.presenter.handle_add_phasequad_button_clicked()
+        self.assertEqual(self.presenter.create_phase_quad_calculation_thread.call_count, 1)
+
+    def test_phasequad_success(self):
+        self.context.group_pair_context.add_pair_to_selected_pairs = mock.Mock()
+        self.presenter.selected_phasequad_changed_notifier.notify_subscribers = mock.Mock()
+        self.presenter._phasequad_obj = MuonPhasequad("test", "table")
+        self.presenter.handle_phasequad_calculation_success()
+
+        self.context.group_pair_context.add_pair_to_selected_pairs.assert_any_call("test_Re_")
+        self.context.group_pair_context.add_pair_to_selected_pairs.assert_any_call("test_Im_")
+        self.presenter.selected_phasequad_changed_notifier.notify_subscribers.assert_any_call({"is_added":True, "name":"test_Re_"})
+        self.presenter.selected_phasequad_changed_notifier.notify_subscribers.assert_any_call({"is_added":True, "name":"test_Im_"})
+
```

Comment:
Worth having a test adding phasequad with a duplicate name both with pairs and another pahesquad. Should handle the unique Real and Imaginary parts as well

Reply:
thats a good idea :)